base/threadingconstructs.jl

```diff
@@ -110,15 +111,73 @@ A barrier is placed at the end of the loop which waits for all tasks to finish
 execution.
 
 The `schedule` argument can be used to request a particular scheduling policy.
-The only currently supported value is `:static`, which creates one task per thread
-and divides the iterations equally among them. Specifying `:static` is an error
-if used from inside another `@threads` loop or from a thread other than 1.
+
+Except for `:static` scheduling, how the iterations are assigned to tasks, and how the tasks
+are assigned to the worker threads are undefined. The exact assignments can be different
+for each execution. The scheduling option should be considered as a hint. The loop body
+code (including any code transitively called from it) must not assume the task and worker
```

Comment:
> must not assume the task

This sounds odd? You generally can't assume which task you are executing on? @tkf what did you mean?

Reply:
With

```julia
@threads for i in 1:Threads.nthreads()
    f(i)
end
```

the user may think that observing different `i` in `f` implies that the different calls of `f(i)` are executed in different tasks (and hence, e.g., blocking communication between `f(1)` and `f(2)` is OK).

Reply:
`must not assume the schedule` or `must not make assumption about the distribution of iterations to tasks`?

Reply:
The latter LGTM

Reply:
```suggestion
code (including any code transitively called from it) must not make assumption about the distribution of iterations to tasks or the worker
```

Reply:
"assumptions", but otherwise +1